src/kyc.ts

```diff
@@ -35,21 +36,32 @@ export class KycApi {
    * perform KYC, or KYB, on the user or business based upon which token is
    * populated: userToken or businessToken.
    */
-  async verify(source: {
+  async verify(type: string, token: string, options: {
     notes?: string,
-    userToken?: string,
-    businessToken?: string,
     manualOverride?: boolean,
     referenceId?: string,
   } = {}): Promise<{
     success: boolean,
     kyc?: Kyc,
     error?: MarqetaError,
   }> {
+    if (!type) {
+      return {
+        success: false,
+        error: mkError('This call requires a verification "type" parameter of ' +
+          '"business" or "user".'),
+      }
+    }
+    let args = { ...options } as Kyc
+    if (type === 'business') {
+      args.businessToken = token
+    } else {
+      args.userToken = token
+    }
     const resp = await this.client.fire('POST',
       'kyc',
       undefined,
-      { ...source }
+      { ...args }
```

Comment:
Why not just have:

```typescript
     if (type === 'business') {
       options.businessToken = token
     } else {
       options.userToken = token
     }
     const resp = await this.client.fire('POST',
       'kyc',
       undefined,
       options as Kyc,
     }
```

Given that `options` is not defined as `const`, it can be changed, and so you can just change it... and go. Now if you were worried about changing the incoming argument, then OK... I get that, and leave it as-is...  but you decide.

Reply:
Because, the `options` parameters is not of type Kyc.  There is no `userToken` or `businessToken` properties on the `options` parameter, so the compiler complains:

`TS2339: Property 'businessToken' does not exist on type '{ notes?: string | undefined; manualOverride?: boolean | undefined; referenceId?: string | undefined; }'.`

That is why `args` is defined as type `Kyc`, and the `userToken` or `businessToken` property is then set based on the `type` parameter.